python/cudf/cudf/tests/conftest.py

```diff
@@ -141,3 +142,17 @@ def pytest_sessionfinish(session, exitstatus):
         del os.environ["_CUDF_TEST_ROOT"]
     except KeyError:
         pass
+
+
+@pytest.fixture(scope="module")
+def default_32bit_integer():
+    cudf.set_option("default_integer_bitwidth", 32)
+    yield
+    cudf.set_option("default_integer_bitwidth", 64)
```

Comment:
Should this be `None` rather than hardcoded to `64`?

Reply:
Don't we actually want to be clear to user what the default bit width is? If you allow `None` for default bitwidth, I assume that conveys that "use whatever inferred dtype anywhere the developer of your function decides to use", which doesn't sounds like a good user experience to me.

Reply:
Using `None` is helpful for reducing API promises. We can document how None is interpreted, but I think it’s fine to use None here.

Reply:
That's fair, this makes three available values for the option: 64, 32 and None.

Reply:
@bdice - Michael and I were pairing on the remaining bits of this PR and we decided not to support `None` as an option right now. As far as we can tell, `None` makes an artificial distinction between Pandas' current behavior and defaulting to 64-bit data types (Pandas seems to always do that anyway).

That being said, if we do discover a case where e.g., Pandas infers data types differently in different places, we can revisit and change the default to `None` (Pandas behaviour).

As for this fixture in particular, I think the following change would suffice:

```python
  def default_32bit_integer():
    default_integer_bitwidth = cudf.get_option("default_integer_bitwidth")
    cudf.set_option("default_integer_bitwidth", 32)
    yield
    cudf.set_option("default_integer_bitwidth", default_integer_bitwidth)
```

Reply:
Since I wrote this comment we discovered a place where defaulting to 64-bit inference _would_ break current code, but it's subtle:

On branch-22.08, the dtype of a Series constructed from a list of numpy scalars is inferred via NumPy:

```python
>>> cudf.Series([np.int8(1), np.int16(1)])
0    1
1    1
dtype: int16
```

In this PR, the default data type will be used:

```python
>>> cudf.set_option("default_integer_bitwidth", 16)
>>> cudf.Series([np.int8(1), np.int16(1)])
0    1
1    1
dtype: int16
```

To prevent breakages in places like this, we decided we should support `None` after all :-)

Sorry for the noise.

Reply:
Wow! That’s a great catch. Goes to show just how complex this logic really is…

Reply:
`None` option is added and tests are updated accordingly.